pkg/runner/run_context.go

```diff
@@ -779,3 +784,37 @@ func (rc *RunContext) localCheckoutPath() (string, bool) {
 	}
 	return "", false
 }
+
+func (rc *RunContext) handleCredentials() (username, password string, err error) {
+	// TODO: remove below 2 lines when we can release act with breaking changes
+	username = rc.Config.Secrets["DOCKER_USERNAME"]
```

Comment:
These secrets are used elsewhere too:
When the runner image that act is using `act -P ubuntu-latest=my.registry/some/image:latest` lives on a private registry that requires auth, act will pull it using the authentication from the secrets.
And since this is `act` internals, I don't think `container.credentials` will be sufficient.
https://github.com/nektos/act/blob/6ebcac37710ae5d4eb87f63f31d7e1be8fe3d217/pkg/container/docker_pull.go#L80-L97
Although I have also created a PR (https://github.com/nektos/act/pull/891) which uses the docker/cli package to read the registry authentication from the local docker config (if it exists).

Reply:
I'm afraid we can't change current behaviour for now